qiime2/tests/test_util.py

```diff
@@ -59,13 +61,25 @@ def test_dst_dir(self):
         with self.assertRaisesRegex(IsADirectoryError, self.dir):
             util.duplicate(self.src, self.dir)
 
-    @mock.patch('qiime2.util.os.link', side_effect=EPERM)
-    def test_perm_error(self,  mocked_link):
-        with self.assertRaisesRegex(PermissionError, "find this"):
+    @mock.patch('qiime2.util.os.link', side_effect=EACCES)
+    def test_perm_error_EACCES(self, mocked_link):
+        with self.assertRaisesRegex(
+                PermissionError, "insufficient r/w permissions"):
             util.duplicate(self.src, self.dst1)
 
         assert mocked_link.called
 
+    @mock.patch('qiime2.util.os.link', side_effect=EPERM)
+    def test_perm_error_EPERM(self, mocked_link):
+        """Does util.duplicate still duplicate if "others" write permissions
+        are absent?"""
```

Comment:
I suspect not, but it also isn't going to be an issue, because duplicate will refuse to overwrite an existing file (by design), so it won't matter what the permissions of the destination are.

Reply:
P.S. this docstring will mask the test-name in most test-runners, which is why our unit-tests usually don't have docstrings.